identity/src/androidTest/java/com/android/identity/IdentityCredentialTest.java

```diff
@@ -0,0 +1,172 @@
+import org.junit.runners.Parameterized.Parameters;
+
+/**
+ * Tests {@link IdentityCredential}.
+ */
+@RunWith(Parameterized.class)
+public class IdentityCredentialTest {
+
+  private static final String CREDENTIAL_NAME = "fake credential for test";
+  private static final int CIPHER_SUITE = CIPHERSUITE_ECDHE_HKDF_ECDSA_WITH_AES_256_GCM_SHA256;
+
+  private IdentityCredentialStore store;
+
+  public IdentityCredentialTest(Object store, Object storeName) {
+    this.store = Preconditions.checkNotNull((IdentityCredentialStore) store);
+    Preconditions.checkNotNull((String) storeName); // only used for test name
+  }
+
+  /**
+   * Tests storing static authentication data of varying size for varying numbers of auth keys.
+   */
+  @MediumTest
+  @Test
+  public void storeStaticAuthenticationData()
+          throws IdentityCredentialException {
+    checkStaticAuthData(1, 100); // 1 key x 100 bytes of static authentication data = 100 Bytes
+    checkStaticAuthData(3, 100);
+    checkStaticAuthData(100, 100); // 100 keys x 100 Bytes = 10 KBytes
+    checkStaticAuthData(1, 100000); // 1 key x 100 KBytes
+  }
+
+  @LargeTest
+  @Test
+  public void storeStaticAuthenticationDataLarge()
+          throws IdentityCredentialException {
+    Assume.assumeTrue("Test fails on API <= 28", VERSION.SDK_INT >= 29 /* Android 10+ */);
+    // Per b/234563696, I've confirmed these failing on:
+    // Pixel 2 API 24
+    // Pixel 2 API 26
+    // Pixel 3 API 28
+    checkStaticAuthData(10, 10000); // fails
+    checkStaticAuthData(100, 1000); // flaky
+    checkStaticAuthData(100, 10000); // fails
+  }
+
+  private void checkStaticAuthData(int numAuthKeys, int staticAuthDataSizeBytes)
+          throws IdentityCredentialException {
+    final int usesPerKey = 3;
+    ProvisioningTest.createCredential(store, CREDENTIAL_NAME);
+    try {
+      IdentityCredential credential = Preconditions.checkNotNull(getCredentialIfPresent());
+
+      storeFakeStaticAuthData(credential, usesPerKey, numAuthKeys, staticAuthDataSizeBytes);
+
+      // Check that this doesn't throw (it did throw in internal b/234563696).
+      try {
+        getCredentialIfPresent();
+      } catch (RuntimeException e) {
+        String msg = String.format(Locale.US,
+                "Failed to load credential for %d authKeys with %d bytes static auth data each",
+                numAuthKeys, staticAuthDataSizeBytes);
+        throw new IdentityCredentialException(msg, e);
+      }
+    } finally {
+      deleteCredentialFromStoreIfPresent();
+    }
+  }
+
+  @Before
+  public void setUp() {
+    deleteCredentialFromStoreIfPresent(); // clean up in case a previous test didn't
+  }
+
+  @After
+  public void tearDown() {
+    deleteCredentialFromStoreIfPresent();
+    store = null;
+  }
+
+  private static void storeFakeStaticAuthData(IdentityCredential credential,
+          int maxUsesPerKey, int numKeys, int staticAuthDataSizeBytes) {
+    Random random = new Random(31337 + numKeys);
```

Comment:
Why bother with `Random` at all?

Why not fill `staticAuthData` with a fixed value?

Reply:
You mean Arrays.fill(staticAuthData, 42) or similar?

That would work, but I generally consider it better practice to use more less regular test data for data that is not regular in practice, just in case the code for some reason behaves notably different from data encountered in the wild (e.g., what if the code fails if and only if a particular byte value occurs in the array?). Random with a fixed seed seems like the easiest way to arrive at such non-regular but deterministic data. No?

I can change it if you feel strongly.

WDYT?